compiler/aarch64/codegen/OMRMemoryReference.cpp

```diff
@@ -850,19 +850,68 @@ uint8_t *OMR::ARM64::MemoryReference::generateBinaryEncoding(TR::Instruction *cu
          }
       else
          {
-         // loadaddrEvaluator() uses addimmx in generateTrgMemInstruction
+         // loadaddrEvaluator() uses addimmx in generateTrg1MemInstruction
          TR_ASSERT(index == NULL, "MemoryReference with unexpected indexed form");
 
-         base->setRegisterFieldRN(wcursor);
-
          if (constantIsUnsignedImm12(displacement))
             {
             *wcursor |= (displacement & 0xfff) << 10; /* imm12 */
+            base->setRegisterFieldRN(wcursor);
             cursor += ARM64_INSTRUCTION_LENGTH;
             }
          else
             {
-            TR_ASSERT_FATAL(false, "Offset is too large for specified instruction.");
+            TR_ASSERT(currentInstruction->getKind() == OMR::Instruction::IsTrg1Mem, "unexpected instruction kind");
+            TR::RealRegister *treg = toRealRegister(((TR::ARM64Trg1MemInstruction *)currentInstruction)->getTargetRegister());
+            uint32_t lower = displacement & 0xffff;
+            uint32_t upper = (displacement >> 16) & 0xffff;
+            bool needSpill = (treg->getRegisterNumber() == base->getRegisterNumber());
+            TR::RealRegister *immreg;
+            TR::RealRegister *stackPtr;
+
+            if (needSpill)
+               {
+               immreg = cg->machine()->getRealRegister(
+                           (base->getRegisterNumber() == TR::RealRegister::x12) ? TR::RealRegister::x11 : TR::RealRegister::x12
```

Comment:
Is there some significance to these particular registers, or are they just arbitrarily chosen?

Reply:
No particular reason.  I chose them from volatile registers that are not used for method arguments.